Fixed Rectangle renders as thin line instead of filled shape for small height

Dhivya-SF4094 opened the pull request:
<!-- Please let the below note in for people that find this PR -->
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

<!--
!!!!!!! MAIN IS THE ONLY ACTIVE BRANCH. MAKE SURE THIS PR IS TARGETING MAIN. !!!!!!! 
-->

### Issue Details:
When height is a small non-integer value (e.g., 1.2), the shape renders incorrectly as a thin stroked line instead of a filled rectangle.
 
### Root Cause
In Shape.cs, the TransformPathForBounds method always adjusts the path bounds for stroke thickness, reducing the dimensions by the StrokeThickness value.
For small heights with the default stroke thickness (1.0px), this resulted in a tiny internal rectangle that appeared as a line rather than a filled shape, unlike BoxView which rendered at the full specified height.
 
### Description of Change
Added a condition to only apply stroke inset when a stroke is actually present (Stroke != null && StrokeThickness > 0). This prevents the shape's interior path from collapsing to an imperceptible height when rendering small filled shapes, particularly when no stroke is applied.
 
### Validated the behaviour in the following platforms
 
- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac
 
### Issues Fixed:
Fixes #31330
 
### Screenshots
| Before  | After |
|---------|--------|
|  <img src="https://github.com/user-attachments/assets/ee6852a6-f7fe-4f66-bc90-68ab3a390e81"> |   <img src="https://github.com/user-attachments/assets/dc70e0b1-84c5-4a43-a835-35ebf8a34b1b">  |
|  <img src="https://github.com/user-attachments/assets/db0e8ba7-19c9-42ea-acf4-a066f6ea236e"> |   <img src="https://github.com/user-attachments/assets/139e4f9a-5a15-4235-99e7-fd650be17371">  |

Dhivya-SF4094 committed:
Fixed Rectangle renders as thin line instead of filled shape for small height values

Dhivya-SF4094 committed:
Fixed Rectangle renders as thin line instead of filled shape for small height values

Dhivya-SF4094 committed:
Updated test sample

Dhivya-SF4094 committed:
Updated test sample

Dhivya-SF4094 commented on src/Controls/tests/TestCases.HostApp/Issues/Issue31330.cs:
```diff
@@ -0,0 +1,32 @@
+            AutomationId = "Issue31330BoxView"
+        };
+
+        var ellipse = new Ellipse
```
@jsuarezruiz Updated the test case to include a Button that changes the StrokeThickness at runtime, starting from 0 and updating to 20 on tap.

jsuarezruiz commented on src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31330.cs:
```diff
@@ -0,0 +1,21 @@
+    public void UpdateSizeOnlyWhenStrokeExists()
+    {
+        App.WaitForElement("Issue31330BoxView");
+        VerifyScreenshot();
```
Test failing on Windows:
```
  at UITest.Appium.HelperExtensions.Wait(Func`1 query, Func`2 satisfactory, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 2530
   at UITest.Appium.HelperExtensions.WaitForAtLeastOne(Func`1 query, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 2547
   at UITest.Appium.HelperExtensions.WaitForElement(IApp app, String marked, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency, Nullable`1 postTimeout) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 743
   at Microsoft.Maui.TestCases.Tests.Issues.Issue31330.UpdateSizeOnlyWhenStrokeExists() in /_/src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31330.cs:line 18
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

jsuarezruiz replied:
Pending snapshot on Mac:
![Uploading image.png…]()

Dhivya-SF4094 replied:
@jsuarezruiz, Added pending snapshots for Android, Mac, and Windows platforms. Re-saved images for failed cases due to recent changes related to shapes.

Dhivya-SF4094 committed:
Resaved image

Dhivya-SF4094 committed:
Resaved image

jsuarezruiz reviewed (changes requested):
Could you rebase and fix the conflicts?

jsuarezruiz reviewed (changes requested):
There are some tests failing that I think could be related with the changes. Example: UpdateSizeOnlyWhenStrokeExists

```
 at UITest.Appium.HelperExtensions.Wait(Func`1 query, Func`2 satisfactory, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 2566
   at UITest.Appium.HelperExtensions.WaitForElement(IApp app, String marked, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency, Nullable`1 postTimeout) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 745
   at Microsoft.Maui.TestCases.Tests.Issues.Issue31330.UpdateSizeOnlyWhenStrokeExists() in /_/src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue31330.cs:line 18
   at System.RuntimeMethodHandle.InvokeMethod(ObjectHandleOnStack target, Void** arguments, ObjectHandleOnStack sig, BOOL isConstructor, ObjectHandleOnStack result)
   at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

Dhivya-SF4094 committed:
Resaved images and updated border test sample

Dhivya-SF4094 committed:
Resaved images and updated border test sample

StephaneDelcroix reviewed (dismissed):
## ✅ Approved

### Summary
This PR correctly fixes Issue #31330 where Rectangle shapes with small heights (e.g., 1.2px) rendered as thin lines instead of filled shapes.

### Root Cause
The `TransformPathForBounds` method in `Shape.cs` always subtracted `StrokeThickness` (default 1.0) from the bounds, even when no `Stroke` was set. For a 1.2px height rectangle: `1.2 - 1.0 = 0.2px` → nearly invisible.

### The Fix
Only apply stroke inset when a stroke is actually present:
```csharp
if (Stroke is not null && StrokeThickness > 0)
{
    // apply inset
}
```

### Changes Reviewed
- **Shape.cs**: Core fix - logically correct ✅
- **BorderTests.cs**: Updated test point calculations to account for shapes now filling their full bounds ✅
- **Issue31330.cs (HostApp)**: Good reproduction test ✅
- **Issue31330.cs (Shared.Tests)**: UI test with correct AutomationId ✅

### Testing
- Rebased cleanly on current main
- AutomationId mismatch between test and HostApp was fixed
- 383 snapshot images updated to reflect the corrected rendering

LGTM! 🚀

PureWeen reviewed (changes requested):
src/Controls/tests/TestCases.HostApp/Issues/Issue31330.cs(101,51): error CS0103: (NETCORE_ENGINEERING_TELEMETRY=Build) The name 'AbsoluteLayoutFlags' does not exist in the current context

Dhivya-SF4094 committed:
Fixed build error and updated test sample

Dhivya-SF4094 committed:
Fixed build error and updated test sample

Dhivya-SF4094 commented:
> src/Controls/tests/TestCases.HostApp/Issues/Issue31330.cs(101,51): error CS0103: (NETCORE_ENGINEERING_TELEMETRY=Build) The name 'AbsoluteLayoutFlags' does not exist in the current context

Fixed the build error in the test sample.
@StephaneDelcroix  Updated the sample layout to avoid scrolling and ensured that both the BoxView and Rectangle are rendered within the visible screen area. Also replaced the red color with green to prevent UI comparison/difference issues during testing.

Dhivya-SF4094 committed:
Addressed AI summary and fixed Border test

Dhivya-SF4094 committed:
Addressed AI summary and fixed Border test

kubaflo reviewed (changes requested):
Could you please resolve conflicts?

Dhivya-SF4094 commented:
> Could you please resolve conflicts?

@kubaflo resolved conflicts.

kubaflo commented:
/azp run maui-pr-uitests , maui-pr-devicetests

kubaflo commented:
The tests are failing. Could you please have a look at them?
